src/features/view-subdomains/SubdomainTable/SubdomainTable.test.tsx

```diff
@@ -0,0 +1,214 @@
+	SubdomainTableRow: (props) => (
+		<tr data-testid="row">
+			<td>{props.zna}</td>
+		</tr>
+	),
+}));
+
+beforeEach(() => {
+	jest.resetAllMocks();
+});
+
+///////////
+// Tests //
+///////////
+
+describe('<SubdomainTable />', () => {
+	beforeEach(() => {
+		mockSubdomainSearch = mockSearchResult;
+		mockInfiniteSubdomains = mockSubdomains;
+	});
+
+	test('should render scroll trigger', () => {
+		const { container } = render(<SubdomainTable zna="0x123" />);
+		expect(container.getElementsByClassName('Trigger')).toHaveLength(1);
+	});
+
+	describe('when browsing subdomains', () => {
+		describe('when toggling between grid and table view', () => {
+			test('should be grid view by default', () => {
+				const { getAllByTestId } = render(<SubdomainTable zna="0x123" />);
+				expect(getAllByTestId('card').length).toBeTruthy();
+			});
+
+			test('should be able to toggle between grid and list views', () => {
+				/*
+				 * @note: this test is flaky, as it relies on the order of the radio buttons.
+				 * Improve this in future iterations!
+				 */
+
+				const { getAllByRole, queryByTestId, getAllByTestId } = render(
+					<SubdomainTable zna="0x123" />,
+				);
+
+				const [listViewButton, gridViewButton] = getAllByRole('radio');
+
+				fireEvent.click(listViewButton);
+
+				expect(getAllByTestId('row').length).toBeTruthy();
+				expect(queryByTestId('card')).toBeNull();
+
+				fireEvent.click(gridViewButton);
+
+				expect(getAllByTestId('card')).toBeTruthy();
+				expect(queryByTestId('row')).toBeNull();
+			});
+		});
+
+		describe('when loading initial subdomains', () => {
+			test('should display correct message: "Loading subdomains"', () => {
+				mockInfiniteSubdomains = {
+					...mockSubdomains,
+					isLoading: true,
+				};
+				const { getByText } = render(<SubdomainTable zna="0x123" />);
+				expect(getByText('Loading subdomains')).toBeInTheDocument();
+			});
+		});
+
+		describe('when loading next page of subdomains', () => {
+			beforeEach(() => {
+				mockInfiniteSubdomains = {
+					...mockSubdomains,
+					isLoading: true,
+					isFetchingNextPage: true,
+				};
+			});
+
+			test('should display correct message: "Loading more subdomains"', () => {
+				const { getByText } = render(<SubdomainTable zna="0x123" />);
+				expect(getByText('Loading more subdomains')).toBeInTheDocument();
+			});
+
+			test('should show already loaded subdomains', () => {
+				const { queryAllByTestId } = render(<SubdomainTable zna="0x123" />);
+				// Checking cards or rows here to make test less flaky
+				const cards = queryAllByTestId('card');
+				const rows = queryAllByTestId('row');
+				expect(cards.length + rows.length).toBe(3);
+			});
+		});
+
+		describe('when subdomains have loaded', () => {
+			test('should display all subdomains', () => {
+				const { queryAllByTestId } = render(<SubdomainTable zna="0x123" />);
+				// Checking cards or rows here to make test less flaky
+				const cards = queryAllByTestId('card');
```

Comment:
I probably would have separated tests into `cards` and `rows` blocks. If there was a breaking change it wouldn't be difficult to identify which of the two is failing but I think it would improve readability. That's more of a personal preference though so no biggie

Reply:
Yeah I was wondering about that, as that's how I initially wrote it. I wanted to try and isolate the `one thing per subdomain` functionality so I rewrote it to not worry about whether it's rendering a card or a row, and instead just worry about how many things it's rendering. That way the test wouldn't break if, for example, the default got changed from card to row.